drivers/floorsense/lockers_websocket.cr

```diff
@@ -0,0 +1,977 @@
+  end
+
+  def delete_rfid(card_number : String)
+    response = post("/restapi/rfid-delete", headers: {
+      "Accept"        => "application/json",
+      "Authorization" => get_token,
+      "Content-Type"  => "application/x-www-form-urlencoded",
+    }, body: URI::Params.build { |form|
+      form.add("csn", card_number)
+    })
+
+    check_success(response)
+  end
+
+  def get_rfid(card_number : String)
+    response = get("/restapi/rfid?csn=#{card_number}", headers: default_headers)
+    parse response, RFID
+  end
+
+  def get_user(user_id : String)
+    existing = @user_cache[user_id]?
+    return existing if existing
+
+    response = get("/restapi/user?uid=#{user_id}", headers: default_headers)
+    user = parse response, User
+    @user_cache[user_id] = user
+    user
+  end
+
+  def user_list(email : String? = nil, name : String? = nil, description : String? = nil, user_group_id : String | Int32? = nil, limit : Int32 = 500, offset : Int32 = 0)
+    query = URI::Params.build { |form|
+      form.add("email", email.not_nil!) if email
+      form.add("name", name.not_nil!) if name
+      form.add("desc", description.not_nil!) if description
+      form.add("ugroupid", user_group_id.to_s) if user_group_id
+      form.add("limit", limit.to_s)
+      form.add("offset", offset.to_s)
+    }
+
+    response = get("/restapi/user-list?#{query}", headers: default_headers)
+    parse response, Array(User)
+  end
+
+  def event_log(codes : Array(String | Int32), event_id : Int64? = nil, after : Int64? = nil, limit : Int32 = 1)
+    query = URI::Params.build { |form|
+      form.add("codes", codes.join(",", &.to_s))
+      form.add("after", after.not_nil!.to_s) if after
+      form.add("event_id", event_id.not_nil!.to_s) if event_id
+      form.add("limit", limit.to_s)
+    }
+
+    response = get("/restapi/event-log?#{query}", headers: default_headers)
+    logs = parse response, Array(LogEntry)
+    logs.sort do |a, b|
+      if a.eventtime == b.eventtime
+        a.eventid <=> b.eventid
+      else
+        a.eventtime <=> b.eventtime
+      end
+    end
+  end
+
+  @[Security(Level::Support)]
+  def clear_user_cache!
+    @user_cache.clear
+  end
+
+  def locate(key : String, controller_id : String? = nil)
+    uri = if controller_id
+            "/restapi/user-locate?cid=#{controller_id}&key=#{URI.encode_www_form key}"
+          else
+            "/restapi/user-locate?name=#{URI.encode_www_form key}"
+          end
+
+    response = get(uri, headers: default_headers)
+    parse response, Array(UserLocation)
+  end
+
+  protected def check_response(resp)
+    check_response(resp, &.not_nil!)
+  end
+
+  protected def check_response(resp)
+    if resp.result
+      yield resp.info
+    else
+      raise "bad response result (#{resp.code}) #{resp.message}"
+    end
+  end
+
+  # ========================================
+  # Lockers Interface
+  # ========================================
+
+  class PlaceLocker
```

Comment:
this is defined in the locatable interface:
https://github.com/PlaceOS/driver/blob/master/src/placeos-driver/interface/lockers.cr#L9

you should use `alias PlaceLocker = PlaceOS::Driver::Interface::Lockers::PlaceLocker`
and can monkey patch it if you want to add some new fields

Similar to what is done in [releezme](https://github.com/PlaceOS/drivers/blob/d82c7bab9fa9fc57d50a816590f075897a19e65a/drivers/vecos/releezme_locations.cr#L10)

Reply:
Currently doing that, if you scroll up you can see the alias.